tests/rptest/tests/datalake/schema_evolution_test.py

```diff
@@ -31,21 +32,93 @@
 
 from ducktape.errors import TimeoutError
 
-QUERY_ENGINES = [
-    QueryEngineType.SPARK,
-    QueryEngineType.TRINO,
-]
 
+# for keeping track of the expected total number of rows across rounds
+# of translation (i.e. calls to _produce)
+class TranslationContext:
+    total: int = 0
+
+
+class ProducerType(str, Enum):
+    AVRO = 'avro'
+    PROTO2 = 'proto2'
+    PROTO3 = 'proto3'
+
+
+class ProtobufVersion(int, Enum):
+    PROTO2 = 2
+    PROTO3 = 3
+
+
+class ProtoProducer:
+    PROTO_TEMPLATE = """
+syntax = "proto{ver}";
 
-class AvroSchema:
+message {name} {{
+{fields}
+}}
+"""
+
+    def __init__(self, redpanda, ver: ProtobufVersion, name: str,
+                 fields: list[dict], topic_name: str):
+        self.redpanda = redpanda
+        self.rpk = RpkTool(redpanda)
+        self.message_name = name
+        p_fields = self._translate_fields(fields)
+        self.schema = self.PROTO_TEMPLATE.format(
+            ver=ver.value,
+            name=self.message_name,
+            fields="\n".join([
+                f"  {'optional ' if ver == ProtobufVersion.PROTO2  else ''}{p_fields[i]['type']} {p_fields[i]['name']} = {i+1};"
+                for i in range(len(p_fields))
+            ]),
```

Comment:
l33t. this would almost have had me grabbing Jinja2 which is available in our deps (i think).

Reply:
yeah lol, this would be totally untenable if the types were any more complicated. it's annoying enough translating between avro & proto & spark & trino that it's probably better to let sleeping dogs lie 🙃

Reply:
Someday this is going to get tricky with nested messages. I think it's worth testing nested messages, but we can do that later, can you make sure there is a Jira for it?

Reply:
yeah no doubt. i'm not totally convinced this is the place for those tests, but yeah I can make a Jira. Apropos of nothing, I suspect (for this and other similar features) that our coverage of various protobuf cases is pretty limited.